core/src/Dscp/Crypto/Hash/Cryptonite.hs

```diff
@@ -42,3 +44,7 @@ BA_INSTANCE_HASH((AbstractSig ss a))
 instance HashAlgorithm algo =>
          HasAbstractHash (CryptoniteFunc algo) LByteString where
     unsafeAbstractHash = AbstractHash . Crypto.hashlazy
+
+instance Hashable (Digest alg) where
+    hash              = hash              . (convert :: Digest alg -> ByteString)
```

Comment:
Hash of the `Digest alg` is the bytestring itself. It _is_ a hash.

Reply:
Nope! `HashMap` doesn't think so.
http://hackage.haskell.org/package/hashable-1.2.7.0/docs/Data-Hashable.html#v:hash

Reply:
Ok, seems right.